main.sh

```diff
@@ -764,6 +766,35 @@ EOF
         register_binfmt qemu-user
     fi
 
+    case "${target}" in
+        aarch64* | arm64*) dpkg_arch=arm64 ;;
+        arm* | thumb*) dpkg_arch=arm ;;
+        i?86-*) dpkg_arch=i386 ;;
+        hexagon-*) dpkg_arch=hexagon ;;
+        loongarch64-*) dpkg_arch=loongarch64 ;;
+        m68k-*) dpkg_arch=m68k ;;
+        mips-* | mipsel-*) dpkg_arch="${target%%-*}" ;;
+        mips64-* | mips64el-*) dpkg_arch="${target%%-*}" ;;
+        mipsisa32r6-* | mipsisa32r6el-*) dpkg_arch="${target%%-*}" ;;
+        mipsisa64r6-* | mipsisa64r6el-*) dpkg_arch="${target%%-*}" ;;
+        powerpc-*spe) dpkg_arch=ppc ;;
+        powerpc-*) dpkg_arch=ppc ;;
+        powerpc64-* | powerpc64le-*) dpkg_arch="${target%%-*}" ;;
+        riscv32*) dpkg_arch=riscv32 ;;
+        riscv64*) dpkg_arch=riscv64 ;;
+        s390x-*) dpkg_arch=s390x ;;
+        sparc-*) dpkg_arch=sparc ;;
+        sparc64-* | sparcv9-*) dpkg_arch=sparc64 ;;
+        x86_64*) dpkg_arch=amd64 ;;
+        *) bail "unrecognized target '${target}'" ;;
+    esac
```

Comment:
I don't actually know what the dpkg_arch should be for most of these... Not sure if this is the right strategy even.